pkg/deploy/util/util.go

```diff
@@ -41,19 +41,10 @@ func LatestDeploymentInfo(config *deployapi.DeploymentConfig, deployments []api.
 // no such deployment. The active deployment is not always the same as the
 // latest deployment.
 func ActiveDeployment(config *deployapi.DeploymentConfig, input []api.ReplicationController) *api.ReplicationController {
-	// we need to create our own copy of the input slice so that our sort here
-	// doesn't change the caller's slice
-	var deployments []api.ReplicationController
-	for _, deployment := range input {
-		deployments = append(deployments, deployment)
-	}
-
-	sort.Sort(ByLatestVersionDesc(deployments))
```

Comment:
There are no guarantees that `input` is sorted and we always need to return the latest complete deployment.